integration/spark-common/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -121,9 +121,11 @@ object DataLoadProcessBuilderOnSpark {
         CarbonProperties.getInstance().getGlobalSortRddStorageLevel()))
     }
 
+    val sortColumnIndex = configuration.getSortColumnRangeInfo.getSortColumnIndex
+
     import scala.reflect.classTag
     val sortRDD = convertRDD
-      .sortBy(_.getData, numPartitions = numPartitions)(RowOrdering, classTag[Array[AnyRef]])
+      .sortBy(_.getKey(sortColumnIndex), numPartitions = numPartitions)(RowOrdering, classTag[Array[AnyRef]])
```

Comment:
Add a comment at line 124 to describe this optimization. It is to reduce the data read for sorting, right?